tests/app/main/test_briefs.py

```diff
@@ -219,39 +229,77 @@ def test_submit_clarification_question_requires_live_brief(self, data_api_client
         assert res.status_code == 404
 
     @mock.patch('app.main.helpers.briefs.send_email')
-    def test_submit_clarification_question_returns_error_page_if_ineligible_supplier_is_not_on_framework(
+    def test_submit_clarification_question_returns_error_page_if_supplier_has_no_services_on_lot(
             self, send_email, data_api_client):
         self.login()
         data_api_client.get_brief.return_value = api_stubs.brief(status='live')
         data_api_client.get_brief.return_value['briefs']['frameworkName'] = 'Digital Outcomes and Specialists'
+        data_api_client.get_brief.return_value['briefs']['lot'] = 'digital-specialists'
         data_api_client.is_supplier_eligible_for_brief.return_value = False
-        data_api_client.get_supplier_framework_info.return_value = {
-            'frameworkInterest': {'onFramework': False}
-        }
+        data_api_client.find_services.side_effect = lambda *args, **kwargs: (
+            {"services": [{"something": "nonempty"}]} if kwargs.get("lot") is None else {"services": []}
+        )
+
         res = self.client.post('/suppliers/opportunities/1/ask-a-question', data={
             'clarification-question': "important question",
         })
+        doc = html.fromstring(res.get_data(as_text=True))
+
         assert res.status_code == 400
-        assert ERROR_MESSAGE_PAGE_HEADING_CLARIFICATION in res.get_data(as_text=True)
-        assert ERROR_MESSAGE_NOT_ON_FRAMEWORK_CLARIFICATION in res.get_data(as_text=True)
+        assert doc.xpath('normalize-space(//h1/text())') == ERROR_MESSAGE_PAGE_HEADING_CLARIFICATION
```

Comment:
Yeah there are two ways I used it - as a "selector" and as a "getter". I preferred the "getter" way as it allowed me to do a regular `==` comparison in python which pytest does handy things with to reveal text diffs etc. There were some places where that wouldn't work and thats where i made do with using it as a selector.